Modules/TRD/src/PulseHeightTrackMatch.cxx

```diff
@@ -0,0 +1,308 @@
+      nTotalTrigger++;
+
+      if (trigger.getNumberOfDigits() == 0) {
+        continue;
+      }
+
+      else {
+        nTrgWDigits++;
+      }
+
+      if (trigger.getNumberOfDigits() > 10000) {
+        mDigitsPerEvent->Fill(9999);
+      } else {
+        mDigitsPerEvent->Fill(trigger.getNumberOfDigits());
+      }
+
+      // now sort digits to det,row,pad
+      std::sort(std::begin(digitsIndex) + trigger.getFirstDigit(), std::begin(digitsIndex) + trigger.getFirstDigit() + trigger.getNumberOfDigits(),
+                [&digitv](unsigned int i, unsigned int j) { return digitIndexCompare_phtm(i, j, digitv); });
+
+      for (const auto& triggerTracks : triggerrecordstracks) {
+
+        if (trigger.getBCData() != triggerTracks.getBCData()) {
+          continue;
+        }
+
+        int nTracks = 0;
+        for (int iTrack = triggerTracks.getFirstTrack(); iTrack < triggerTracks.getFirstTrack() + triggerTracks.getNumberOfTracks(); iTrack++) {
+          nTracks++;
+          const auto& track = tracks[iTrack];
+
+          // printf("Got track with %i tracklets \n", track.getNtracklets());
+          int ntracklets = track.getNtracklets();
+          if (ntracklets > 10)
+            mTrackletsPerMatchedTrack->Fill(9);
+          else
+            mTrackletsPerMatchedTrack->Fill(ntracklets);
+
+          for (int iLayer = 0; iLayer < 6; iLayer++) {
+            int trackletIndex = track.getTrackletIndex(iLayer);
+
+            if (trackletIndex == -1) {
+              continue;
+            }
+
+            const auto& trklt = tracklets[trackletIndex];
+
+            int det = trklt.getDetector();
+            int sec = det / 30;
+            int row = trklt.getPadRow();
+            // int col = trklt.getPadCol();
+
+            // obtain pad number relative to MCM center
+            int padLocal = trklt.getPositionBinSigned() * o2::trd::constants::GRANULARITYTRKLPOS;
+            // MCM number in column direction (0..7)
+            int mcmCol = (trklt.getMCM() % o2::trd::constants::NMCMROBINCOL) + o2::trd::constants::NMCMROBINCOL * (trklt.getROB() % 2);
+            int col = 6.f + mcmCol * ((float)o2::trd::constants::NCOLMCM) + padLocal;
+
+            for (int iDigit = trigger.getFirstDigit() + 1; iDigit < trigger.getFirstDigit() + trigger.getNumberOfDigits() - 1; ++iDigit)
+
+            {
+              const auto& digit = digits[iDigit];
+              const auto& digitBefore = digits[iDigit - 1];
+              const auto& digitAfter = digits[iDigit + 1];
+
+              std::tuple<unsigned int, unsigned int, unsigned int> aa, ba, ca;
+              aa = std::make_tuple(digitBefore.getDetector(), digitBefore.getPadRow(), digitBefore.getPadCol());
+              ba = std::make_tuple(digit.getDetector(), digit.getPadRow(), digit.getPadCol());
+              ca = std::make_tuple(digitAfter.getDetector(), digitAfter.getPadRow(), digitAfter.getPadCol());
```

Comment:
I do not see the need for intermediate tuples. The values can be calculated when needed.
It's also not necessary to process the adjacent pads at this point, because for now you are only looking at the coordinates of the central pad. Once you found a match for that, you can check the neighbors.